src/leappto/actor_cli.py

```diff
@@ -0,0 +1,154 @@
+    """
+    host_port, sep, container_port = arg.partition(":")
+    host_port = int(host_port)
+    if not sep:
+        container_port = host_port
+    else:
+        container_port = int(container_port)
+    return str(host_port), container_port
+
+
+def to_port_map(items):
+    port_map = {}
+    for source, target in items:
+        port_map[source] = port_map.get(source, []) + [target]
+    return {k: list(set(v)) for k, v in port_map.items()}
+
+
+def _path_spec(arg):
+    path = os.path.normpath(arg)
+    if not os.path.isabs(path):
+        raise ValueError("Path '{}' is not absoulute or valid.".format(str(arg)))
+
+    return path
+
+
+def make_argument_parser():
+    ap = ArgumentParser()
+    ap.add_argument('-v', '--version', action='version', version=VERSION, help='display version information')
+    parser = ap.add_subparsers(help='sub-command', dest='action')
+    migrate_cmd = parser.add_parser('migrate-machine', help='migrate source VM to a target container host')
+
+    migrate_cmd.add_argument('machine', help='source machine to migrate')
+    migrate_cmd.add_argument('-t', '--target', default='localhost', help='target VM name')
+    migrate_cmd.add_argument(
+        '--tcp-port',
+        default=None,
+        dest="forwarded_tcp_ports",
+        nargs='*',
+        type=_port_spec,
+        help='(Re)define target tcp ports to forward to macrocontainer - [target_port:source_port]'
+    )
+    migrate_cmd.add_argument(
+        '--no-tcp-port',
+        default=None,
+        dest="excluded_tcp_ports",
+        nargs='*',
+        type=_port_spec,
+        help='define tcp ports which will be excluded from the mapped ports [[target_port]:source_port>]'
+    )
+
+    migrate_cmd.add_argument(
+        '--exclude-path',
+        default=None,
+        dest="excluded_paths",
+        nargs='*',
+        type=_path_spec,
+        help='define paths which will be excluded from the source'
+    )
+    migrate_cmd.add_argument("--ignore-default-port-map", default=False,
+                             help='Default port mapping detected by leapp toll will be ignored', action="store_true")
+    migrate_cmd.add_argument('--container-name', '-n', default=None,
+                             help='Name of new container created on target host')
+    migrate_cmd.add_argument(
+        '--force-create',
+        action='store_true',
+        help='force creation of new target container, even if one already exists'
+    )
+    migrate_cmd.add_argument('--disable-start', dest='disable_start', default=False,
+                             help='Migrated container will not be started immediately', action="store_true")
+    _add_identity_options(migrate_cmd, context='source')
+    _add_identity_options(migrate_cmd, context='target')
+
+    return ap
+
+
+def make_base_string(s):
+    return {"value": s}
+
+
+def make_base_bool(s):
+    return {"value": s}
+
+
+def start_agent_if_not_available():
+    if 'SSH_AUTH_SOCK' in os.environ:
```

Comment:
not sure we want to limit only to var value here, i would rather try to access that agent with current SSH_AUTH_SOCK and SSH_AGENT_PID other wise i think this check is not very efficent

Reply:
well you might have a point here - However this is already a curtesy option to do that, and to be quite honest I don't even would want to support that in the end - I will drop the identity file support - I never really liked it anyway because we don't even know if it is password protected or not so rather let the user deal with the 'not being able to access the remote machine' scenario